### PR TITLE
Add TraceConfig and TraceParams in SDK.

### PR DESCRIPTION
**Pull request by songy23**
Fixes https://github.com/open-telemetry/opentelemetry-java/issues/335.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
See comments in https://github.com/open-telemetry/opentelemetry-java/issues/279. I think now that config is no longer part of the API we can make it part of the Tracer. What do you (others) think?

File: sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceParams.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import com.google.common.base.Preconditions;
+import io.opentelemetry.trace.Event;
+import io.opentelemetry.trace.Link;
+import io.opentelemetry.trace.Span;
+import javax.annotation.concurrent.Immutable;
+
+/** Class that holds global trace parameters. */
+@AutoValue
+@Immutable
+public abstract class TraceParams {
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceParams.java]: We can name this TraceConfig if we go with my proposal. Also see all the fields in https://github.com/open-telemetry/opentelemetry-proto/blob/master/src/opentelemetry/proto/trace/v1/trace_config.proto#L37

Comment (songy23) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceParams.java]: Sounds good, consolidated in https://github.com/open-telemetry/opentelemetry-java/pull/336/commits/b6c3859e78704d3dd802285855b78a0cf5407103.

Other configs are added in https://github.com/open-telemetry/opentelemetry-java/pull/336/commits/c8574082521b60015e5549aef63324b504070e1a.

File: sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceConfig.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.samplers.Samplers;
+import javax.annotation.concurrent.Immutable;
+
+/** Class that holds global trace parameters. */
+@AutoValue
+@Immutable
+public abstract class TraceConfig {
+  // These values are the default values for all the global parameters.
+  // TODO: decide which default sampler to use
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceConfig.java]: Probably an issue for this is worth it.

Comment (songy23) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceConfig.java]: Filed #347 for further discussions.

**Pull request merged**